### PR TITLE
use Context>>#homeMethod

### PR DESCRIPTION
**Pull request by MarcusDenker**
- use #homeMethod in Context #methodNode and #methodSelector
- simplify printOn: #selector already does the nil check and fallback to default
- use homeMethod in explicitRequirement and BytecodeEncoder

### CODE REVIEW

File: src/Kernel/Object.class.st [hunk 1]
```diff
@@ -894,9 +894,9 @@ Object >> explicitRequirement [
 	| originalMethod originalArguments errorBlock originalReceiver callingContext originalSelector |
 	errorBlock := [ ^ self error: 'Explicitly required method' ].
 	callingContext := thisContext sender.
-	originalMethod := callingContext compiledCode method.
+	originalMethod := callingContext homeMethod.
 	originalMethod isFromTrait
-		ifFalse: errorBlock.
+		ifFalse: [ errorBlock value].
```
Comment (privat) [inline src/Kernel/Object.class.st:899]: For curiosity, why this change?

Comment (MarcusDenker) [reply src/Kernel/Object.class.st:899]: just to use the optized ifFalse... I guess not important here as speed does not matter
